modules/users/server/controllers/users/users.password.server.controller.js

```diff
@@ -32,7 +32,7 @@ exports.forgot = function (req, res, next) {
         User.findOne({
           username: req.body.username.toLowerCase()
         }, '-salt -password', function (err, user) {
-          if (!user) {
+          if (err || !user) {
```

Comment:
Shouldn't we be handling an `err` differently than when we don't find a User?

Our handling of Mongoose errors seems to be a pain point in our coverage. It's difficult (and I don't think we have done it anywhere) to write tests against possible Mongoose errors; other than predictable schema validation errors. So when we write something like `if (err) {...}` when no validation errors are possible (as in this case), we lose coverage.

Maybe it's something we can work on. Although, I'm not sure the exact strategy we should employ. I just feel like when we don't find a User, and when Mongoose throws an unexpected error, are two different cases that should be handled independently.

WDYT?